proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -547,6 +547,532 @@ locale delete_one = delete_one_conc + delete_one_abs +
                (invs' and cte_at' (cte_map ptr))
           (cap_delete_one ptr) (cteDeleteOne (cte_map ptr))"
 
+(** from here **)
+
+(* for instantiations; only need to specify what you are adding *)
+lemma corres_inst_add:
+  "corres_underlying sr nf nf' r (P and Q) (P' and Q') f g
+   \<Longrightarrow> corres_underlying sr nf nf' r (P and Q) (P' and Q') f g" .
+
+(** sr_inv **)
```

Comment:
As always, naming is difficult. I’d like something short, but I am not fully convinced with this... any ideas?